### PR TITLE
gh: check copyrights and make Apache only approved

### PR DESCRIPTION
**Pull request by kikofernandez**
Adds the checks on copyrights, and makes Apache the only accepted license.
Anything outside Apache should be flagged and approved in the Security panel.

### CODE REVIEW

File: scripts/scan-code.escript [hunk 1]
```diff
@@ -38,18 +38,22 @@ cli() ->
...
 approved() ->
-    [ ~"mit", ~"agpl-3.0", ~"apache-2.0", ~"boost-1.0", ~"llvm-exception",
-      ~"lgpl-2.1-plus", ~"cc0-1.0", ~"bsd-simplified", ~"bsd-new", ~"pcre",
-      ~"fsf-free", ~"autoconf-exception-3.0", ~"mpl-1.1", ~"public-domain",
-      ~"autoconf-simple-exception", ~"unicode", ~"tcl", ~"gpl-2.0 WITH classpath-exception-2.0",
-      ~"zlib", ~"lgpl-2.0-plus WITH wxwindows-exception-3.1", ~"lgpl-2.0-plus",
-      ~"openssl-ssleay", ~"cc-by-sa-3.0", ~"cc-by-4.0", ~"dco-1.1", ~"fsf-ap",
-      ~"agpl-1.0-plus", ~"agpl-1.0", ~"agpl-3.0-plus", ~"classpath-exception-2.0",
-      ~"ietf-trust"].
+    [ <<"apache-2.0">> ].
```
Comment (kikofernandez) [inline scripts/scan-code.escript:41]: We only accept Apache

File: scripts/scan-code.escript [hunk 2]
```diff
@@ -38,18 +38,22 @@ cli() ->
...
+
+reviewed() ->
+    [ <<"mit">>, <<"boost-1.0">>, <<"llvm-exception">>,
+      <<"cc0-1.0">>, <<"bsd-simplified">>, <<"bsd-new">>, <<"pcre">>,
+      <<"fsf-free">>, <<"autoconf-exception-3.0">>, <<"public-domain">>,
+      <<"autoconf-simple-exception">>, <<"unicode">>, <<"tcl">>, <<"gpl-2.0 WITH classpath-exception-2.0">>,
+      <<"zlib">>, <<"lgpl-2.0-plus WITH wxwindows-exception-3.1">>,
+      <<"openssl-ssleay">>, <<"cc-by-sa-3.0">>, <<"cc-by-4.0">>, <<"dco-1.1">>, <<"fsf-ap">>,
+      <<"classpath-exception-2.0">>, <<"ietf-trust">> ].
```
Comment (kikofernandez) [inline scripts/scan-code.escript:50]: in cases of auto-generated code or special cases, we could review contributions under these licenses, but these must be very special cases.
The idea is that the automated check marks these for review, and we accept them on an ongoing basis. 
No PR should be merged without its corresponding license approval

File: scripts/scan-code.escript [hunk 3]
```diff
@@ -38,18 +38,22 @@ cli() ->
...
 
 not_approved() ->
-    [~"gpl", ~"gpl-3.0-plus", ~"gpl-2.0", ~"gpl-1.0-plus", ~"unlicense",
-     ~"erlangpl-1.1", ~"gpl-2.0-plus", ~"null", 'null'].
+    [<<"gpl">>, <<"gpl-3.0-plus">>, <<"gpl-2.0">>, <<"gpl-1.0-plus">>, <<"unlicense">>,
+     <<"lgpl-2.0-plus">>, <<"lgpl-2.1-plus">>, <<"agpl-1.0-plus">>, <<"agpl-1.0">>,
+     <<"agpl-3.0-plus">>, <<"erlangpl-1.1">>, <<"gpl-2.0-plus">>, <<"null">>, <<"agpl-3.0">>,
+     <<"mpl-1.1">>, 'null'].
```
Comment (kikofernandez) [inline scripts/scan-code.escript:56]: we do not approve some licenses, and I mess up the list initially

**Comment by kikofernandez** [comment]
this is a meta error. the introduction of the LICENSE rule to correctly change AND rules by ORs cannot be applied to itself, so it reports a license compliance error when it is ok.

**Comment by garazdawi** [comment]
The change seems to crash when run on a push and not a pr: https://github.com/kikofernandez/otp/actions/runs/11958065364/job/33336711990

**Comment by kikofernandez** [comment]
[like]  Kiko Fernandez-reyes A reacted to your message:
________________________________

**Review by garazdawi — COMMENTED** [review]
As we are reporting copyright and license information now we need to remove any duplicate files from the SARIF artifact array. For example like this:

```erlang
                 ~"artifacts" =>
                     [ #{
                         ~"location" => #{
                                          ~"uri" => File
                                         },
                         ~"length" => -1
                        } || File <- lists:usort([F || {_, F, _} <- Errors])]
                     ],
```

The error shows up here: https://github.com/kikofernandez/otp/actions/runs/11973269459/job/33382224159

**Comment by kikofernandez** [comment]
Thanks, I was thinking that it was ok to report the same file for copyright issues and for license issues, but that should not be the case for the sarif generation, since files should be unique. Thanks again! Change pushed

**Review by garazdawi — APPROVED** [review]

**Pull request merged**